### PR TITLE
Replace isbinaryfile with istextorbinary

### PR DESCRIPTION
**Pull request by Toilal**
This avoid reading the file twice, and also drop one dependency when #737 is merged.

compared to [isbinaryfile](https://github.com/gjtorikian/isBinaryFile), [istextorbinary](https://github.com/bevry/istextorbinary) can handles a `Buffer` as input instead of filename.

### CODE REVIEW

File: lib/actions/actions.js [hunk 1]
```diff
@@ -35,12 +35,14 @@ actions._prepCopy = function _prepCopy(source, destination, process) {
...
-  var binary = isBinaryFile(source);
-  if (!binary) {
-    encoding = 'utf8';
-  }
+  var binary = true;
+
+  body = fs.readFileSync(source);
 
-  body = fs.readFileSync(source, encoding);
+  if (istextorbinary.isTextSync(undefined, body)) {
```
Comment (SBoudrias) [inline lib/actions/actions.js]: Let's not change value of a variable inside the `if` statement. It's making it harder to reason about.

```js
isText = istextorbinary.isTextSync(undefined, body)

if (isText) {
  // foo
}

// etc
```

Comment (Toilal) [inline lib/actions/actions.js]: well, it doesn't change any value, but i'll extract it np.

Comment (SBoudrias) [inline lib/actions/actions.js]: It changes `var binary` value. You should get rid of this variable.

It is easier to think about a thing that doesn't change than mutable variable values.

File: lib/actions/actions.js [hunk 2]
```diff
@@ -35,17 +34,18 @@ actions._prepCopy = function _prepCopy(source, destination, process) {
...
 
-  body = fs.readFileSync(source, encoding);
+  if (isText) {
+    body = body.toString();
 
-  if (typeof process === 'function' && !binary) {
-    body = process(body, source, destination, {
-      encoding: encoding
-    });
+    if (typeof process === 'function') {
```
Comment (Toilal) [inline lib/actions/actions.js]: It's not really part of this PR, but why this is not called for binary data ?

Comment (SBoudrias) [inline lib/actions/actions.js]: That's mostly legacy feature. You'll have the capacity to edit binary content using the new file system (package is mem-fs-editor).

**Comment by SBoudrias** [comment]
LGTM thanks for taking care of it!

**Comment by Toilal** [comment]
A pleasure, great to have responsive people like your team in this project :)
